main/src/vtr_radar_lidar/src/modules/localization/localization_icp_module.cpp

```diff
@@ -150,7 +150,16 @@ void LocalizationICPModule::run_(QueryCache &qdata0, OutputCache &,
     aligned_map_normals_mat = T_s_m * map_normals_mat;
 
     // project to 2D
-    aligned_map_mat.row(2).setZero();
+    for (uint j = 0; j < aligned_map_mat.cols(); ++j) {
+        const double px = aligned_map_mat(0, j);
+        const double py = aligned_map_mat(1, j);
+        const double pz = aligned_map_mat(2, j);
+        const double rho = std::sqrt(px * px + py * py + pz * pz);
+        const double phi = std::atan2(py, px);
+        aligned_map_mat(0, j) = rho * std::cos(phi);
+        aligned_map_mat(1, j) = rho * std::sin(phi);
+        aligned_map_mat(2, j) = 0.0;
+    }
```

Comment:
This is not the a normal projection into the plane z=0. Points with a large z value will be stretched along the radius. Is this what you want?

Reply:
yup

Reply:
The previous section restricts the lidar points to those within the radar's field of view:

https://github.com/utiasASRL/vtr3/blob/ce0f3fbfc72a6d6a228ea1a921704d32cea293db/main/src/vtr_radar_lidar/src/modules/localization/localization_icp_module.cpp#L108-L135

In the config file, we set the elevation range to +-0.05rad which is 3 x the vertical field of view of the radar.

Due to the nonzero vertical beam divergence of the radar, this "spherical" projection is a more accurate way of aligning lidar points with the 2D navtech radar data.